### PR TITLE
load all types from pandas

### PR DESCRIPTION
**Pull request by hendrik-obj**
more data types support when loading from pandas with `from_pandas`
* if `materialization == 'cte'` all data types are supported that Bach supports
* If `materialization == 'table'` supported dtypes are : `supported_pandas_dtypes`

### CODE REVIEW

File: bach/bach/from_pandas.py [hunk 1]
```diff
@@ -176,26 +181,34 @@ def _from_pd_shared(
...
+                else:
+                    raise TypeError(f'multiple types found in column {column}: {types}')
+            else:
+                raise TypeError(f'unsupported dtype for {column}: {dtype}')
+
+        if column == index:
+            index_dtypes[str(column)] = dtype
+        else:
+            dtypes[str(column)] = dtype
+
```
[the remark this answers is not in the document]
Comment (hendrik-obj) [reply bach/bach/from_pandas.py]: much better

File: bach/tests/functional/bach/test_df_from_pandas.py [hunk 2]
```diff
@@ -26,6 +29,20 @@
...
+            [1, 1.324, True, datetime.datetime(2021, 5, 3, 11, 28, 36, 388), 'Ljouwert', datetime.date(2021, 5, 3),
+             ['Sûkerbôlle'], UUID('36ca4c0b-804d-48ff-809f-28cf9afd078a'), {'a': 'b'}, datetime.date(2021, 5, 3)],
+            [2, 2.734, True, datetime.datetime(2021, 5, 4, 23, 28, 36, 388), 'Snits', datetime.date(2021, 5, 4),
+             ['Dúmkes'], UUID('81a8ace2-273b-4b95-b6a6-0fba33858a22'), {'c': ['d', 'e']}, ['Dúmkes']],
+            [3, 3.52, False, datetime.datetime(2022, 5, 3, 14, 13, 13, 388), 'Drylts', datetime.date(2022, 5, 3),
+             ['Grutte Pier Bier'], UUID('8a70b3d3-33ec-4300-859a-bb2efcf0b188'), {'f': 'g', 'h': 'i'}, ['Grutte Pier Bier']]
+        ]
+TYPES_COLUMNS = ['int_column', 'float_column', 'bool_column', 'datetime_column',
+       'string_column', 'date_column', 'list_column', 'uuid_column',
+       'dict_column', 'mixed_column']
```
[the remark this answers is not in the document]
Comment (hendrik-obj) [reply bach/tests/functional/bach/test_df_from_pandas.py]: good one, added it to the test.

**Review by KathiaBarahona — APPROVED** [review]
lgtm

File: bach/bach/from_pandas.py [hunk 3]
```diff
@@ -58,7 +59,7 @@ def from_pandas_store_table(engine: Engine,
         * append: Insert new values to the existing table.
     """
     # todo add dtypes argument that explicitly let's you set the supported dtypes for pandas columns
-    df_copy, index_dtypes, dtypes = _from_pd_shared(df, convert_objects)
+    df_copy, index_dtypes, dtypes = _from_pd_shared(df, convert_objects, cte=False)
```
Comment (jansentom) [inline bach/bach/from_pandas.py:62]: shouldn't we use the `sql_models.model.Materialization` type for this?

**Pull request merged**